### PR TITLE
Update train.py

### PR DESCRIPTION
**Pull request by Dante-Berth**
Since Gymnasium 1.0.0, final_info does not exist anymore, i updated the code.

### CODE REVIEW

**Comment by ShaneFlandermeyer** [comment]
Just now looking at this.  Gonna have to change the sequential buffer class to make this work as it currently assumes each sub-environment gets a new transition tuple on each call to insert().

**Comment by Dante-Berth** [comment]
Why not using : https://github.com/instadeepai/flashbax ? I did not get your point ? Could you explain a bit more

**Comment by Dante-Berth** [comment]
Have you tried using jax.lax.scan instead of loops? I tried it, but I didn’t push forward with it because I found that TD MPC was slower

**Comment by ShaneFlandermeyer** [comment]
Yeah, we've found that the speedup is negligible compared to regular loops since we're only doing a few iterations in all the loops. Plus the non-scanned version is much easier to parse for someone digging into the code.

**Comment by ShaneFlandermeyer** [comment]
Also, I believe I've fixed the replay buffer implementation to make this work. Testing a bit more in the develop branch to double check that there are no performance regressions. 

To answer your question about existing buffer libraries, I decided to make my own for a few reasons: 
* The buffer implementation for this algorithm is extremely simple and imo it's not worth adding a new dependency for it. 
* I have found that dejax and flashbax support multi-step trajectories and parallel environments, but doing both requires some hacks or padding
* The original repo uses TorchRL's buffers. These would obviously slow things down a bunch.

**Comment by ShaneFlandermeyer** [comment]
Ok, I have tested and verified the performance with the new replay buffer and version 1.0.0. The changes will live in the develop branch for a bit, but feel free to test them out and let me know if there are any other problems!
